### ISSUE TITLE
support lower-case vendor prefixes in styleNames

### ISSUE THREAD

**Issue by jbonta** [facebook/react#1063]
TLDR: Let's allow `style={msTransform: '...', webkitTransition: '...'}`

for stylenames, we [recommend](http://facebook.github.io/react/tips/inline-styles.html) that JavaScript keys start with a capital letter for vendor prefixes, presumably because [hyphenation](https://github.com/facebook/react/blob/63d6cc013e3dfffdc7c924f9d0309c73e471aa74/src/vendor/core/hyphenate.js#L32) is then trivial (so they start with a single leading hyphen). This forces a user to put a vendor-specific key that isn't even valid, or it produces css that [doesn't](http://jsfiddle.net/wkLBM/) really [work](http://jsfiddle.net/n7gxp/).

We probably need to re-work this, or build awareness of vendor-specific prefixes into the core of react's style building. This comes up [here](https://github.com/facebook/react/commit/5abcce534382d85887f3d33475e8e54e3b5d8457#diff-c524cc735426e792db75807b0d78e742R59) too.
